### PR TITLE
SS4 upgrade

### PR DESCRIPTION
**Pull request by alex-dna**

### CODE REVIEW

File: src/extensions/RhinoAssessmentHierarchyExtension.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+	public function getAccount()
+	{
+		return $this->owner->getParentOfType(RhinoAccount::class);
+	}
+
+	public function getParentOfType($type)
+	{
+		$parents = $this->owner->getBreadcrumbItems();
+		$parents->remove($this);
```
Comment (djmattski) [inline src/extensions/RhinoAssessmentHierarchyExtension.php]: @alex-dna Remove($this->owner)? Just not sure if that context is needed for this operation.

File: src/extensions/RhinoAssessmentHierarchyExtension.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+class RhinoAssessmentHierarchyExtension extends DataExtension
+{
+
+	/**
+	 * Helper methods
+	 */
+	public function getModule()
+	{
+		return $this->owner->getParentOfType(RhinoModule::class);
```
Comment (djmattski) [inline src/extensions/RhinoAssessmentHierarchyExtension.php:19]: @alex-dna I don't know enough about this module - but if we are saying that they can use any nesting structure in the readme - I see that these could return null. Is that handled?

File: src/extensions/RhinoAssessmentHierarchyExtension.php (hunk 1 above)
Comment (alex-dna) [reply src/extensions/RhinoAssessmentHierarchyExtension.php]: Not sure what you mean. This code is on an extension, not sure how to call that method other than on the owner...

File: src/extensions/RhinoAssessmentHierarchyExtension.php (hunk 2 above)
Comment (alex-dna) [reply src/extensions/RhinoAssessmentHierarchyExtension.php:19]: These methods are use on the RhinoAssessment, and the lack of results should be handled there.

File: src/extensions/RhinoAssessmentHierarchyExtension.php (hunk 1 above)
Comment (djmattski) [reply src/extensions/RhinoAssessmentHierarchyExtension.php]: Yeah calling the method from the owner is correct - but the passed param is the extension. Should remove() take the owner rather? 
i.e. $parents->remove($this->owner)

Comment (alex-dna) [reply src/extensions/RhinoAssessmentHierarchyExtension.php]: Oh, sorry I wasn't ready the right line. You are absolutely right.

**Pull request merged**